platform/security/idp/security-idp-server/src/test/java/org/codice/ddf/security/idp/server/IdpEndpointTest.java

```diff
@@ -849,6 +852,48 @@ public void testPassiveLoginPkiUnsupported()
     assertThat(responseStr, containsString("status:RequestUnsupported"));
   }
 
+  @Test
+  public void testPassiveLoginPkiUnsupportedBinding()
+      throws SecurityServiceException, WSSecurityException, CertificateEncodingException,
+          IOException {
+    String samlRequest = authNRequestPassivePkiGetUnsupportedBinding;
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    X509Certificate x509Certificate = mock(X509Certificate.class);
+
+    Subject subject = mock(Subject.class);
+    PrincipalCollection principalCollection = mock(PrincipalCollection.class);
+    SecurityAssertion securityAssertion = mock(SecurityAssertion.class);
+    SecurityToken securityToken = mock(SecurityToken.class);
+    SecurityManager securityManager = mock(SecurityManager.class);
+    when(subject.getPrincipals()).thenReturn(principalCollection);
+    when(principalCollection.asList()).thenReturn(Collections.singletonList(securityAssertion));
+    when(securityAssertion.getSecurityToken()).thenReturn(securityToken);
+    // this mock element is what will cause the signature error
+    when(securityToken.getToken()).thenReturn(mock(Element.class));
+    when(securityManager.getSubject(anyObject())).thenReturn(subject);
+    idpEndpoint.setSecurityManager(securityManager);
+    idpEndpoint.setStrictSignature(false);
+
+    when(request.isSecure()).thenReturn(true);
+    when(request.getRequestURL()).thenReturn(requestURL);
+    when(request.getAttribute(ContextPolicy.ACTIVE_REALM)).thenReturn("*");
+    // dummy cert
+    when((X509Certificate[]) request.getAttribute(requestCertificateAttributeName))
+        .thenReturn(new X509Certificate[] {x509Certificate});
+    when(x509Certificate.getEncoded()).thenReturn(new byte[48]);
+
+    Response response =
+        idpEndpoint.showGetLogin(samlRequest, relayState, signatureAlgorithm, signature, request);
+    String responseStr =
+        StringUtils.substringBetween(
+            response.getEntity().toString(), "SAMLResponse=", "&RelayState");
```

Comment:
Nifty 👍

Reply:
I'll approve this, but it's a bit fragile and leaks a detail from the implementation. There's no requirement that we add query parameters in the particular order we do now and no guarantee that we won't find need to insert another param in the future that might break this test.